src/nvim/os/env.c

```diff
@@ -223,6 +223,13 @@ void init_homedir(void)
 #endif
     homedir = xstrdup(var);
   }
+  else {
+    char IObuff_signed[MAXPATHL] = "";
+    if (uv_os_homedir(IObuff_signed, (size_t *)MAXPATHL)) {
```

Comment:
`MAXPATHL` can't be passed in here.  As noted in the docs for `uv_os_homedir()`, it will change the passed in value to report how large the string actually is:

> On success size is set to the string length of buffer. On UV_ENOBUFS failure size is set to the required length for buffer, including the null byte.

A local variable should be used instead.